src/borg/crypto.pyx

```diff
@@ -551,98 +550,102 @@ cdef class CHACHA20_POLY1305(_CHACHA_BASE):
 
 cdef class AES:
     """A thin wrapper around the OpenSSL EVP cipher API - for legacy code, like key file encryption"""
+    cdef CIPHER cipher
     cdef EVP_CIPHER_CTX *ctx
-    cdef int is_encrypt
-    cdef unsigned char iv_orig[16]
+    cdef unsigned char *enc_key
+    cdef int cipher_blk_len
+    cdef int iv_len
+    cdef unsigned char iv[16]
     cdef long long blocks
 
-    def __cinit__(self, is_encrypt, key, iv=None):
+    def __init__(self, enc_key, iv=None):
+        assert isinstance(enc_key, bytes) and len(enc_key) == 32
+        self.enc_key = enc_key
```

Comment:
Maybe do a rename sweep later (avoiding more conceptual/naming changes in this PR).

That being said a better naming convention may be cipher_key, mac_key and id_key.